File: Data Structures & Algorithms/is-palindrome/submission-0.py
import re
class Solution:
    def isPalindrome(self, s: str) -> bool:
        i =0 
        s = s.split()
        s = ''.join(s)
        s = s.lower()
        s = re.sub(r'[^a-z0-9]','',s)
        print(s)
        while i<len(s)/2:
            if s[i]!=s[-1*(i+1)]:
                #print(s[i], s[-1*i], s[i+1])
                return False
            i +=1
        return True

        

File: Data Structures & Algorithms/rotate-matrix/submission-0.java
class Solution {
    public void rotate(int[][] matrix) {
        int n = matrix.length;
        int m = matrix[0].length;

        for(int i=0;i<n/2;i++)
        {
            for(int j=i; j<m-i-1;j++)
            {
                System.out.println(i + "" + j);
                int k=0;
                int x = i;
                int y = j;
                int sub = matrix[i][j];
                while(k<=2)
                {
                    matrix[x][y] = matrix[m-y-1][x];
                    System.out.println(x + " = x, " + "y =" + y + ", m ="+matrix[x][y] );
                    int s = x;
                    x = m-y-1;
                    y = s;
                    k++;
                }
                matrix[x][y] =sub;
            }
        }
    }
}
